# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# T2 SDE: package/.../tzdata/destdir.patch
# Copyright (C) 2013 The T2 SDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./Makefile.orig	2012-11-13 03:34:49.000000000 +0100
+++ ./Makefile	2012-12-26 11:31:32.025712966 +0100
@@ -277,6 +277,7 @@
 cc=		cc
 CC=		$(cc) -DTZDIR=\"$(TZDIR)\"
 
+RANLIB=		ranlib
 TZCSRCS=	zic.c localtime.c asctime.c scheck.c ialloc.c
 TZCOBJS=	zic.o localtime.o asctime.o scheck.o ialloc.o
 TZDSRCS=	zdump.c localtime.c ialloc.c
@@ -318,29 +319,30 @@
 
 install:	all $(DATA) $(REDO) $(TZLIB) $(MANS) $(TABDATA)
 		$(ZIC) -y $(YEARISTYPE) \
-			-d $(TZDIR) -l $(LOCALTIME) -p $(POSIXRULES)
-		-rm -f $(TZDIR)/iso3166.tab $(TZDIR)/zone.tab
-		cp iso3166.tab zone.tab $(TZDIR)/.
-		-mkdir $(TOPDIR) $(ETCDIR)
-		cp tzselect zic zdump $(ETCDIR)/.
-		-mkdir $(TOPDIR) $(MANDIR) \
-			$(MANDIR)/man3 $(MANDIR)/man5 $(MANDIR)/man8
-		-rm -f $(MANDIR)/man3/newctime.3 \
-			$(MANDIR)/man3/newtzset.3 \
-			$(MANDIR)/man5/tzfile.5 \
-			$(MANDIR)/man8/tzselect.8 \
-			$(MANDIR)/man8/zdump.8 \
-			$(MANDIR)/man8/zic.8
-		cp newctime.3 newtzset.3 $(MANDIR)/man3/.
-		cp tzfile.5 $(MANDIR)/man5/.
-		cp tzselect.8 zdump.8 zic.8 $(MANDIR)/man8/.
+			-d $(DESTDIR)$(TZDIR) -p $(POSIXRULES)
+		ln -sf ..$(TZDIR)/$(LOCALTIME) $(DESTDIR)/etc/localtime
+		-rm -f $(DESTDIR)$(TZDIR)/iso3166.tab $(DESTDIR)$(TZDIR)/zone.tab
+		cp iso3166.tab zone.tab $(DESTDIR)$(TZDIR)/.
+		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(ETCDIR)
+		cp tzselect zic zdump $(DESTDIR)$(ETCDIR)/.
+		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) \
+			$(DESTDIR)$(MANDIR)/man3 $(DESTDIR)$(MANDIR)/man5 $(DESTDIR)$(MANDIR)/man8
+		-rm -f $(DESTDIR)$(MANDIR)/man3/newctime.3 \
+			$(DESTDIR)$(MANDIR)/man3/newtzset.3 \
+			$(DESTDIR)$(MANDIR)/man5/tzfile.5 \
+			$(DESTDIR)$(MANDIR)/man8/tzselect.8 \
+			$(DESTDIR)$(MANDIR)/man8/zdump.8 \
+			$(DESTDIR)$(MANDIR)/man8/zic.8
+		cp newctime.3 newtzset.3 $(DESTDIR)$(MANDIR)/man3/.
+		cp tzfile.5 $(DESTDIR)$(MANDIR)/man5/.
+		cp tzselect.8 zdump.8 zic.8 $(DESTDIR)$(MANDIR)/man8/.
 
 INSTALL:	ALL install date.1
-		-mkdir $(TOPDIR) $(BINDIR)
-		cp date $(BINDIR)/.
-		-mkdir $(TOPDIR) $(MANDIR) $(MANDIR)/man1
-		-rm -f $(MANDIR)/man1/date.1
-		cp date.1 $(MANDIR)/man1/.
+		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(BINDIR)
+		cp date $(DESTDIR)$(BINDIR)/.
+		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) $(DESTDIR)$(MANDIR)/man1
+		-rm -f $(DESTDIR)$(MANDIR)/man1/date.1
+		cp date.1 $(DESTDIR)$(MANDIR)/man1/.
 
 version.h:
 		(echo 'static char const PKGVERSION[]="($(PACKAGE)) ";' && \
@@ -357,10 +359,10 @@
 		chmod +x yearistype
 
 posix_only:	zic $(TDATA)
-		$(ZIC) -y $(YEARISTYPE) -d $(TZDIR) -L /dev/null $(TDATA)
+		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR) -L /dev/null $(TDATA)
 
 right_only:	zic leapseconds $(TDATA)
-		$(ZIC) -y $(YEARISTYPE) -d $(TZDIR) -L leapseconds $(TDATA)
+		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR) -L leapseconds $(TDATA)
 
 # In earlier versions of this makefile, the other two directories were
 # subdirectories of $(TZDIR).  However, this led to configuration errors.
@@ -372,9 +374,9 @@
 # You must replace all of $(TZDIR) to switch from not using leap seconds
 # to using them, or vice versa.
 other_two:	zic leapseconds $(TDATA)
-		$(ZIC) -y $(YEARISTYPE) -d $(TZDIR)-posix -L /dev/null $(TDATA)
+		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)/posix -L /dev/null $(TDATA)
 		$(ZIC) -y $(YEARISTYPE) \
-			-d $(TZDIR)-leaps -L leapseconds $(TDATA)
+			-d $(DESTDIR)$(TZDIR)/right -L leapseconds $(TDATA)
 
 posix_right:	posix_only other_two
 
@@ -383,10 +385,10 @@
 zones:		$(REDO)
 
 $(TZLIB):	$(LIBOBJS)
-		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
+		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(LIBDIR)
+		ar ru $(DESTDIR)$@ $(LIBOBJS)
 		if [ -x /usr/ucb/ranlib ] || [ -x /usr/bin/ranlib ]; \
-			then ranlib $@ ; fi
+			then $(RANLIB) $(DESTDIR)$@ ; fi
 
 date:		$(DATEOBJS)
 		$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(DATEOBJS) $(LDLIBS)
